components/RichText.tsx

```diff
@@ -0,0 +1,114 @@
+import { documentToReactComponents } from "@contentful/rich-text-react-renderer";
+import { BLOCKS, MARKS, INLINES } from "@contentful/rich-text-types";
+import React from "react";
+import styles from "styles/RichText.module.scss";
+
+type Theme = "LIGHT" | "DARK" | "INDIGO";
+
+type RichTextProps = {
+    theme: Theme;
+    content: any;
+};
+
+const renderParagraph = (theme: Theme, children: React.ReactNode) => {
+    switch (theme) {
+        case "LIGHT":
+            return <p className={styles.paragraphLight}>{children}</p>;
+        case "DARK":
+            return <p className={styles.paragraphLight}>{children}</p>;
+        case "INDIGO":
+            return <p className={styles.paragraphLight}>{children}</p>;
+    }
+};
+
+const renderLink = (theme: Theme, children: React.ReactNode) => {
+    switch (theme) {
+        case "LIGHT":
+            return <p className={styles.paragraphLight}>{children}</p>;
+        case "DARK":
+            return <p className={styles.paragraphLight}>{children}</p>;
+        case "INDIGO":
+            return <p className={styles.paragraphLight}>{children}</p>;
+    }
+};
```

Comment:
To be deleted or refactored (I avoided this solution initially as I would have to pass the `node` with an unknown type into the function - however, we could use this function and just pass the node with `: any` instead